internals-js/src/operations.ts

```diff
@@ -648,10 +676,6 @@ export class SelectionSet extends Freezable<SelectionSet> {
   }
 
   collectUsedFragmentNames(collector: Map<string, number>) {
-    if (!this.fragments) {
```

Comment:
Note to reviewers: that condition (and the same below) was currently incorrect because `this.fragments` is not always set at all the levels of a `SelectionSet`. We could probably change this and using `this.fragments` is set at all the levels, but that's a bit easy to get wrong so it feels that not relying on that was less fragile.